### ISSUE TITLE
Recipe train config about seq2seq_tds

### ISSUE THREAD

**Issue by zjuturtle** [facebookresearch/wav2letter#773]
I've tried to reproduce seq2seq_tds. Following the guide wav2letter/recipes/models/seq2seq_tds/librispeech/READEME.md. 
But I got lr=0 at second epoch.

```
I0731 10:19:50.247529 78586 Train.cpp:345] epoch:        1 | nupdates:            4 | lr: 0.046652 | lrcriterion: 0.046652 | runtime: 00:00:14 | bch(ms): 3723.39 | smp(ms): 137.43 | fwd(ms): 1769.15 | crit-fwd(ms): 638.87 | bwd(ms): 987.09 | optim(ms): 483.00 | loss:  267.12091 | train-LER:  0.00 | train-WER:  0.00 | dev-clean-loss:  288.92285 | dev-clean-LER: 943.12 | dev-clean-WER: 593.36 | dev-other-loss:  260.87748 | dev-other-LER: 1096.28 | dev-other-WER: 674.13 | avg-isz: 890 | avg-tsz: 029 | max-tsz: 063 | hrs:    0.32 | thrpt(sec/sec): 76.53
Memory Manager Stats
MemoryManager type: CachingMemoryManager
Number of allocated blocks:254
Size of free block pool (small):33
Size of free block pool (large):164
Total native mallocs:168
Total native frees:0
I0731 10:19:51.299285 78586 Train.cpp:730] Finished window pretraining.
I0731 10:19:51.301118 78586 Train.cpp:566] Shuffling trainset
I0731 10:19:51.307119 78586 Train.cpp:573] Epoch 2 started!
I0731 10:48:19.034971 78586 Train.cpp:345] epoch:        2 | nupdates:         8792 | lr: 0.000000 | lrcriterion: 0.000000 | runtime: 00:28:07 | bch(ms): 191.95 | smp(ms): 5.77 | fwd(ms): 44.40 | crit-fwd(ms): 3.81 | bwd(ms): 126.88 | optim(ms): 13.91 | loss:  287.89709 | train-LER: 102.24 | train-WER: 105.63 | dev-clean-loss:  180.50355 | dev-clean-LER: 100.00 | dev-clean-WER: 100.00 | dev-other-loss:  159.25201 | dev-other-LER: 100.00 | dev-other-WER: 100.00 | avg-isz: 1229 | avg-tsz: 042 | max-tsz: 097 | hrs:  960.56 | thrpt(sec/sec): 2049.79
...
```

I noice there is a gamma(=0.5) parameter in `train.cfg`. And the code associated with gamma in `Train.cpp` is here
```cpp

      if (FLAGS_lrcosine) {
      //we do not use lrcosine so forget it
          const double pi = std::acos(-1);
          lrScheduleScale =
              std::cos(((double)curBatch) / ((double)nbatches) * pi / 2.0);
        } else {
       // lrScheduleScale will quickly drop to zero because curBatch increase. 
          lrScheduleScale =
              std::pow(FLAGS_gamma, (double)curBatch / (double)FLAGS_stepsize);
        }
      // the second epoch will just set lr to nearly zero
        netopt->setLr(
            initlr * lrDecayScale * lrScheduleScale *
            std::min(curBatch / double(FLAGS_warmup), 1.0));
        critopt->setLr(
            initcritlr * lrDecayScale * lrScheduleScale *
            std::min(curBatch / double(FLAGS_warmup), 1.0));
```
The recipe use librispeech(960hours in total) as trainset which contains 281241 files(or we say samples) and will have 17578 batches(batchsize=16, each batch has 16 samples). The stepsize=40. So the `curBatch` in code will increase quickly.
Maybe it should use curEpoch? So the code should be
```
std::pow(FLAGS_gamma, (double)curEpoch / (double)FLAGS_stepsize);
```
I'm not sure it is a bug or not.

**Comment by xuqiantong** [comment]
Hey, this is due to a series of recent changes to the LR scheduler, which changed the behaviors of several flags. Specifically, regarding to your use case, the `--pretrainWindow ` and `--stepsize` are now referring to the number of updates, rather than epochs (as in the recipe). So, please try to set them larger accordingly. For example, if you have N updates per epoch, you should multiply both flags by N.

**Comment by zjuturtle** [comment]
solved it
